src/executor.ts

```diff
@@ -106,6 +112,19 @@ export class Executor {
     });
   }
 
+  /**
+   * Execute the root script.
+   */
+  async execute(): Promise<Result<unknown, Failure[]>> {
+    const result = await this.getExecution(this._rootConfig).execute();
+    // Wait for services to shut down.
+    // TODO(aomarks) In watch mode, directly-invoked scripts (and the services
+    // they depend on) should not block here, since they should continue
```

Comment:
Is this always the case? Should some services restart in watch mode depending onhow they use their dependencies and files?

Reply:
In watch mode, the directly invoked services will get passed off to the next iteration's execution, which will decide whether to shut down or preserve the child processes.